app/src/main/cpp/loop.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "common.h"
+
+namespace zen::display_system::oculus {
+
+class Loop {
+ public:
+  struct ISource;
+
+  DISABLE_MOVE_AND_COPY(Loop);
+  Loop(struct android_app *app) : app_(app) {}
+  ~Loop() = default;
+
+  void Run();
+  void Terminate();
+
+  /* Add a busy loop sources to be processed at each loop iteration */
+  void AddBusy(std::weak_ptr<Loop::ISource> source);
```

Comment:
Use weak_ptr to avoid circular reference.

- `Loop` refers `ISource`s (weak_ptr)
- `ISource` can possess `Loop` (shared_ptr)